src/_guides/language/language-tour.md

```diff
@@ -3323,34 +3204,28 @@ as E, T, S, K, and V.
 
 ### Why use generics?
 
-Because types are optional in Dart 1.x, you never *have* to use generics.
-You might *want* to, though, for the same reason you might want to use
-other types in your code: types (generic or not) let you document and
-annotate your code, making your intent clearer.
-
-{% include optional-types-2.0.html %}
+Generics are often required for type safety, but they have other benefits
+than just allowing your code to run:
 
-{% comment %}
-update-for-dart-2
-{% endcomment %}
+* Properly specifying generic types results in better generated code.
+* You can use generics to reduce code duplication.
 
-For example, if you intend for a list to contain only strings, you can
+If you intend for a list to contain only strings, you can
 declare it as `List<String>` (read that as “list of string”). That way
-you, your fellow programmers, and your tools (such as your IDE and
-the Dart VM in checked mode) can detect that assigning a non-string to
+you, your fellow programmers, and your tools can detect that assigning a non-string to
 the list is probably a mistake. Here’s an example:
 
 <?code-excerpt "misc/test/language_tour/generics_test.dart (why-generics)"?>
 {% prettify dart %}
 var names = new List<String>();
 names.addAll(['Seth', 'Kathy', 'Lars']);
-names.add(42); // Fails in checked mode (succeeds in production mode).
+names.add(42); // Error.
```

Comment:
[nit] Drop the period:

-  // Error. --> // Error

(I believe that this change was only done in this file, not the original source.)